src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -2826,6 +2826,146 @@ namespace cryptonote
     typedef epee::misc_utils::struct_init<response_t> response;
   };
 
+  #define KV_SERIALIZE_ENTRY_FIELD_IF_REQUESTED(var) \
+  if (this_ref.requested_fields.var || !this_ref.requested_fields.explicitly_set) KV_SERIALIZE(var)
+
+  struct COMMAND_RPC_GET_N_SERVICE_NODES
+  {
+
+    // Boolean values indicate whether corresponding
+    // fields should be included in the response
+    struct requested_fields_t {
+
+      bool explicitly_set = false;
```

Comment:
Could you annotate this with LOKI_RPC_DOC_INTROSPECT and add comments alongside this field like it is done in the response.

You can just add one line for explicitly_set and mention that the rest is enabling the respective fields in Service Node Info and forgo comments for the remainder of the fields.

This way the documentation generator can go ahead and pick it up.

Reply:
`explicitly_set` is not supposed to be set by the user (it is not serialised/deserialised), but I added to comment to say so. I've added LOKI_RPC_DOC_INTROSPECT as well. The comment to `requested_fields_t` already mentions that the boolean fields control which fields to include.